extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesAndWorkerTaskRunnerFactory.java

```diff
@@ -33,7 +34,11 @@ public class KubernetesAndWorkerTaskRunnerFactory implements TaskRunnerFactory<K
 
   private final KubernetesTaskRunnerFactory kubernetesTaskRunnerFactory;
   private final HttpRemoteTaskRunnerFactory httpRemoteTaskRunnerFactory;
```

Comment:
Should this also be injected as a Provider since it is not needed when the remote task runner factory is configured?

Reply:
Changed `HttpRemoteTaskRunnerFactory` to be injected as Provider.